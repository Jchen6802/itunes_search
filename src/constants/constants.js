import Immutable from 'immutable';

export const SET_ALBUMS = 'SET_ALBUMS';
export const SET_STATUS = 'SET_STATUS';
export const UPDATE_SEARCH_TERM = 'UPDATE_SEARCH_TERM';
export const FETCH_SOURCE = 'https://itunes.apple.com/search';
export const STATUSES = {
    ERROR: 'ERROR',
    INITIAL: 'INITIAL',
    SEARCHING: 'SEARCHING',
};

export const INITIAL_STATE = {
    albums: Immutable.List(),
    status: STATUSES.INITIAL,
    searchTerm: ''
};
