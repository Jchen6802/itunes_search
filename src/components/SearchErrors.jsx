import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import styled from 'styled-components';
import React, { Component } from 'react';

import { statusSelector } from '../selectors';
import { STATUSES } from '../constants/constants';

const ErrorsContainer = styled.div`
    color: red;
    margin-top: 20px;
`;

export class SearchErrors extends Component {
    render () {
        return this.inErrorState() && <ErrorsContainer>
            Searcher encountered an error! Please try searching again.
        </ErrorsContainer>;
    }

    inErrorState = () => {
        return this.props.status === STATUSES.ERROR;
    }
}

SearchErrors.propTypes = {
    status: PropTypes.string.isRequired
};

const mapStateToProps = (state) => {
    return {
        status: statusSelector(state)
    }
};

export default connect(mapStateToProps)(SearchErrors);
