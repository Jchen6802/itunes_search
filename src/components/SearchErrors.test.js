import React from 'react';
import renderer from 'react-test-renderer';

import { STATUSES } from '../constants/constants';
import { SearchErrors } from './SearchErrors.jsx';

it('renders as expected when in the error status', () => {
    const searchErrors = renderer.create(
        <SearchErrors status={STATUSES.ERROR}/>
    ).toJSON();

    expect(searchErrors).toMatchSnapshot();
});

it('does not render when not in the error status', () => {
    const searchErrors = renderer.create(
        <SearchErrors status={STATUSES.INITIAL}/>
    ).toJSON();

    expect(searchErrors).toMatchSnapshot();
});
