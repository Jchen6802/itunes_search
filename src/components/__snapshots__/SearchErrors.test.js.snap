// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`does not render when not in the error status 1`] = `null`;

exports[`renders as expected when in the error status 1`] = `
<div
  className="sc-bdVaJa OItRm"
>
  Searcher encountered an error! Please try searching again.
</div>
`;
